narwhal/executor/src/notifier.rs

```diff
@@ -56,6 +56,9 @@ impl<State: ExecutionState + Send + Sync + 'static> Notifier<State> {
                     )
                     .await;
             }
+            self.metrics
+                .batch_execution_latency
+                .observe(batch.metadata.timestamp.elapsed().as_secs_f64());
```

Comment:
It is possible to create (even across validators) batches with same digests and basically overwrite existing data. That might skew results as when we fetch a batch we can't really know if a newer version (hence one with newer timestamp) has been written. Ultimately we would like to include the batch timestamps in the header it self, but considered it for now a bit far fetched. If suspicious results are observed then might go the extra mile.

Reply:
Actually the timestamp might be a relatively good source of entropy to include in the batch digest to ensure that two different batches with same transaction data will not have the same digest (just some thoughts)